### PR TITLE
✨ #73 RDPC Admin Util

### PR DESCRIPTION
**Pull request by demariadaniel**
Adds Util to check for RDPC Write Permissions

https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/gh/icgc-argo/ego-token-utils/73

### CODE REVIEW

File: test/utils.test.ts [hunk 1]
```diff
@@ -77,6 +77,12 @@ describe('isRdpcMember', () => {
   });
 });
 
+describe('isRdpcAdmin', () => {
+  it('should return false if failed', () => {
+    expect(validator.isRdpcAdmin(validator.getPermissionsFromToken('ssdfg'))).toBe(false);
+  });
+});
+
```
Comment (demariadaniel) [inline test/utils.test.ts:85]: Need updated JWT w/ RDPC Admin permissions for positive test

Comment (joneubank) [reply test/utils.test.ts:85]: I can generate this for you and send over.

File: src/argoRoleChecks.ts [hunk 2]
```diff
@@ -57,6 +57,15 @@ export const isRdpcMember = (permissions: string[]): boolean => {
...
 
+/**
+ * check if a given set of permissions has rdpc write access
+ * @param permissions
+ */
+export const isRdpcAdmin = (permissions: string[]): boolean =>
+  permissions
+    .filter(policy => policy && policy.startsWith(RDPC_PREFIX))
+    .some(code => code.includes(PERMISSIONS.WRITE));
+
```
Comment (joneubank) [inline src/argoRoleChecks.ts:68]: I think we need a way to check if you are admin of a specific RDPC. Maybe thats a separate method like `canWriteToRdpc(permissions: string[], rdpcCode: string): boolean` which can match for a specific rdpc name (like we have with programs).

**Review by joneubank — CHANGES_REQUESTED** [review]

File: src/argoRoleChecks.ts (hunk 2 above)
Comment (demariadaniel) [reply src/argoRoleChecks.ts:68]: Essentially the same as this: https://github.com/icgc-argo/ego-token-utils/pull/75#discussion_r1279875531

File: test/utils.test.ts (hunk 1 above)
Comment (joneubank) [reply test/utils.test.ts:85]:
```
eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2MjE3ODkzMzgsImV4cCI6MjA1Mjc2NTc2Nywic3ViIjoiZjAzYWMxZGQtY2RmYS00NzRiLTlmYTEtNGMzYzFmNDNhYjBlIiwiaXNzIjoiZWdvIiwiYXVkIjpbXSwianRpIjoiNTg1YzAxYzYtNDUwOS00ZTk3LTg2NWEtMmMwNzZkY2M4OGEzIiwiY29udGV4dCI6eyJzY29wZSI6WyJSRFBDLUNBLldSSVRFIl0sInVzZXIiOnsiZW1haWwiOiJqb2hubnkuYWRtaW5AZXhhbXBsZS5jb20iLCJzdGF0dXMiOiJBUFBST1ZFRCIsImZpcnN0TmFtZSI6IkpvaG5ueSIsImxhc3ROYW1lIjoiUkRQQyBBZG1pbiIsImNyZWF0ZWRBdCI6MTYyMTc4OTMzODAwMCwibGFzdExvZ2luIjoxNjIxOTc2NTM4MDAwLCJwcmVmZXJyZWRMYW5ndWFnZSI6bnVsbCwidHlwZSI6IlVTRVIiLCJwcm92aWRlclR5cGUiOiJHT09HTEUiLCJwcm92aWRlclN1YmplY3RJZCI6ImRhY29BZG1pbjEyMzQifX19.eFna5LLa9K4GdX55iIkmIviK4vU-v4SZGfpBu8po6MOyj7CDi30piqHMGFz0C0dHj6a0Hb7RxOPzEKC2MoMHbvc6A4JwH9IucOb9YW6SD_Qi99Y6QaJoUWMimj-CWzYEgv6V78gFZSN5i1PRoizxfvZRIawTSp_iO2kl6TWpyctQ8P2SIed3riJ6E1Z6uVFp4m3Hzgj3ZAc519vFIu2q2rCnDyb2C-tgzcxGPE6uKnn60MksYYg_qngQe8W5icSbYNjtQhc87V9CwFaZNtmWaKPc59axCapfKcR7hjBBQvSqx5sWq9UZ77oYyAWQYhThcUJSNuUTG18eS8SUBDReFw
```

Comment (demariadaniel) [reply test/utils.test.ts:93]: Updated w/ token, added positive/negative test cases

File: src/argoRoleChecks.ts (hunk 2 above)
Comment (demariadaniel) [reply src/argoRoleChecks.ts:68]: OK I've added a `canWriteToRdpc` util: https://github.com/icgc-argo/ego-token-utils/pull/76/commits/1cf5f73d79dca9e6ea4d2ed11b1b4f9d6c0a62ad

Looking at it I'm not sure how necessary `isRdpcAdmin` is now; that's basically saying 'can write to ANY RDPC'

**Review by joneubank — APPROVED** [review]

**Comment by joneubank** [comment]
I will rebase this on the latest version of bug/72 PR which includes the Jenkins changes.

**Pull request merged**